apps/jsonrpc2/lib/jsonrpc2/spec_handler.ex

```diff
@@ -1,5 +1,114 @@
 defmodule JSONRPC2.SpecHandler do
   use JSONRPC2.Server.Handler
 
+  alias Blockchain.Block
+  alias Blockchain.Blocktree
+  alias ExthCrypto.Hash.Keccak
+  alias ExWire.PeerSupervisor
+  alias ExWire.Sync
+  # web3 Methods
+
   def handle_request("web3_clientVersion", _), do: Application.get_env(:jsonrpc2, :mana_version)
+  def handle_request("web3_sha3", [param = "0x" <> _]), do: Keccak.kec(param)
+
+  # net Methods
+  def handle_request("net_version", _), do: Application.get_env(:ex_wire, :network_id)
+  def handle_request("net_listening", _), do: Application.get_env(:ex_wire, :discovery)
+  def handle_request("net_peerCount", _), do: PeerSupervisor.connected_peer_count()
+
+  # eth Methods
+  def handle_request("eth_protocolVersion", _), do: {:error, :not_supported}
+
+  def handle_request("eth_syncing", _) do
+    sync_state = get_last_sync_state()
+
+    current_block = get_last_sync_block(sync_state)
+
+    %{
+      currentBlock: current_block,
+      startingBlock: sync_state.starting_block_number,
+      highestBlock: sync_state.highest_block_number
+    }
+  end
+
+  def handle_request("eth_coinbase", _), do: {:error, :not_supported}
```

Comment:
This is great. I like that we can add these slowly one by one while also clearly seeing what is missing 👍